DS and Algorithm/BST.py

```diff
@@ -0,0 +1,34 @@
+class TreeNode:
+    def __init__(self, key):
+        self.left = None
+        self.right = None
+        self.val = key
+
+def inorder_traversal(root):
+    result = []
+    if root:
+        # Traverse the left subtree
+        result += inorder_traversal(root.left)
+
+        # Visit the root node
+        result.append(root.val)
+
+        # Traverse the right subtree
+        result += inorder_traversal(root.right)
+
+    return result
+
+# Example usage
+if __name__ == "__main__":
+    # Creating a sample Binary Search Tree
+    root = TreeNode(5)
+    root.left = TreeNode(3)
+    root.right = TreeNode(7)
+    root.left.left = TreeNode(2)
+    root.left.right = TreeNode(4)
+    root.right.left = TreeNode(6)
+    root.right.right = TreeNode(8)
+
+    print("Inorder Traversal of BST:")
+    result = inorder_traversal(root)
+    print(result)
```

Comment:
**Code Structure Issue**: The class TreeNode and its methods are well defined, but the indentation is inconsistent with Python's PEP 8 standards which recommend 4 spaces per indentation level. <br> **Fix**: Adjust the indentation to use 4 spaces consistently throughout the TreeNode class and its methods to adhere to PEP 8 standards. <br> **Code Suggestion**: 
 ```
 class TreeNode:
     def __init__(self, key):
         self.left = None
         self.right = None
         self.val = key

 def inorder_traversal(root):
     result = []
     if root:
         # Traverse the left subtree
         result += inorder_traversal(root.left)

         # Visit the root node
         result.append(root.val)

         # Traverse the right subtree
         result += inorder_traversal(root.right)

     return result

 # Example usage
 if __name__ == "__main__":
     # Creating a sample Binary Search Tree
     root = TreeNode(5)
     root.left = TreeNode(3)
     root.right = TreeNode(7)
     root.left.left = TreeNode(2)
     root.left.right = TreeNode(4)
     root.right.left = TreeNode(6)
     root.right.right = TreeNode(8)

     print("Inorder Traversal of BST:")
     result = inorder_traversal(root)
     print(result)
 ```